node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs

```diff
@@ -768,29 +855,114 @@ mod tests {
+            .gas_used(gas_used)
+            .status(status)
+            .build();
         let _blockchain_client_server = MBCSBuilder::new(port)
-            .response("0x3B9ACA00".to_string(), 0)
-            .response("0xFFF0".to_string(), 0)
-            .response(
-                "0x000000000000000000000000000000000000000000000000000000000000FFFF".to_string(),
-                0,
+            .begin_batch()
+            .err_response(
+                429,
+                "The requests per second (RPS) of your requests are higher than your plan allows."
+                    .to_string(),
+                7,
             )
+            .raw_response(r#"{ "jsonrpc": "2.0", "id": 1, "result": null }"#.to_string())
+            .response("trash".to_string(), 0)
+            .raw_response(tx_receipt_response_not_present)
+            .raw_response(tx_receipt_response_failed)
+            .raw_response(tx_receipt_response_success)
+            .end_batch()
             .start();
+        let subject = make_blockchain_interface_web3(port);
 
-        let expected_err_factory = |wallet: &Wallet| {
-            BlockchainAgentBuildError::TransactionID(
-                wallet.address(),
-                BlockchainError::QueryFailed(
-                    "Transport error: Error(IncompleteMessage) for wallet 0x0000…6364".to_string(),
-                ),
-            )
-        };
+        let result = subject
+            .process_transaction_receipts(tx_hash_vec)
+            .wait()
+            .unwrap();
 
-        build_of_the_blockchain_agent_fails_on_blockchain_interface_error(
-            port,
-            expected_err_factory,
+        assert_eq!(result[0], TransactionReceiptResult::Error("RPC error: Error { code: ServerError(429), message: \"The requests per second (RPS) of your requests are higher than your plan allows.\", data: None }".to_string()));
+        assert_eq!(result[1], TransactionReceiptResult::NotPresent);
+        assert_eq!(
+            result[2],
+            TransactionReceiptResult::Error(
+                "invalid type: string \"trash\", expected struct Receipt".to_string()
+            )
+        );
+        assert_eq!(result[3], TransactionReceiptResult::NotPresent);
+        assert_eq!(
+            result[4],
+            TransactionReceiptResult::TransactionFailed(TransactionReceipt {
+                transaction_hash: tx_hash_5,
+                transaction_index: Default::default(),
+                block_hash: None,
+                block_number: None,
+                cumulative_gas_used: U256::from(0),
+                gas_used: None,
+                contract_address: None,
+                logs: vec![],
+                status: Some(status_failed),
+                root: None,
+                logs_bloom: H2048::default()
+            })
+        );
+        assert_eq!(
+            result[5],
+            TransactionReceiptResult::Found(TransactionReceipt {
```

Comment:
Are you guys sure that we want to operate with these massive data structures from Web3. We held a discussion and I thought you might try to replace it with something closely ours with just as much in it as what we will utilize.